### ISSUE TITLE
Metabase Lineage and URN casing

### ISSUE THREAD

**Issue by cbeauch** [datahub-project/datahub#8814]
**Describe the bug**

In Metabase, there are two different types of "charts". Ones that have been created in the UI and ones that have been created by writing SQL. The metabase connector behaves differently depending on the two and the issue relates to the function `get_datasource_urn` in `metabase.py`. 

If the Metabase chart is created with native SQL, the else statement will be hit.`LineageRunner` then converts the query to lowercase, thus we end up with a lowercase URN. In my case, my BigQuery URN's are not lowercase, so lineage never matches.

LineageRunner is only used in the else, so if the question is created in the Metabase UI, the casing does not get converted.

I've also had to hard-code my GCP project ID in `metabase.py` because if their is a dash in the project-id, it is completely dropped... but that is another problem. 

tl;dr: dataset URN is cased differently depending on the type of Metabase chart (native sql vs UI based). LineageRunner is to blame

**Comment by cbeauch** [comment]
**IF** my BigQuery recipe contained `convert_urns_to_lowercase`, then I could fix this issue myself by going

```
source_tables.add(
    f"{source_db + '.' if source_db else 'hardcoded-hyphen-project.'}{source_schema.lower()}.{source_table.lower()}"
)
```

(assuming I only use one gcp project)

**Comment by tsafacjo** [comment]
can I take it ?

**Comment by hsheth2** [comment]
@tsafacjo sorry we didn't see your message here, but would love for you to take a stab at it!

**Comment by giorgiobevilacqua** [comment]
Hi! It seems to be still an issue in v 0.12.1

**Comment by hsheth2** [comment]
Note that this might be fixed by https://github.com/datahub-project/datahub/pull/9714
